src/logging/log_manager.cpp

```diff
@@ -0,0 +1,146 @@
+#include "duckdb/logging/log_manager.hpp"
+#include "duckdb/logging/log_storage.hpp"
+#include "duckdb/main/database.hpp"
+#include "duckdb/main/client_context.hpp"
+#include "duckdb/main/client_data.hpp"
+
+namespace duckdb {
+
+unique_ptr<Logger> LogManager::CreateLogger(LoggingContext context, bool thread_safe, bool mutable_settings) {
+	// Make a copy of the config holding the lock
+	LogConfig config_copy;
+	{
+		unique_lock<mutex> lck(lock);
+		config_copy = config;
```

Comment:
Do we need to make this copy?

Reply:
I remember the reason now: in the constructor of a Logger, a Logger needs to register itself to the LogManager which requires locking the LogManager. See for example `ThreadSafeLogger(LogConfig &config_p, LoggingContext &context_p, LogManager &manager)`.  I did this to avoid deadlocking. However this is indeed not super clean.

I will just add another constructor that takes a RegisteredLogContext instead